mitmproxy/addons/tlsconfig.py

```diff
@@ -225,6 +228,11 @@ def tls_start_client(self, tls_start: tls.TlsData) -> None:
                 http2=ctx.options.http2,
             )
         )
+
+        # Trigger the hook registered by third-party addons.
+        if self.tls_start_client_cb:
+            self.tls_start_client_cb(tls_start)
```

Comment:
I agree. I add the hook just because I want to reuse the code in addons/tlsconfig, and it might not a good idea. It looks like some duplication is inevitable. 😔